chain-vote/src/tally.rs

```diff
@@ -18,11 +21,36 @@ pub type ProofOfCorrectShare = CorrectElGamalDecrZkp;
 /// Common Reference String
 pub type Crs = GroupElement;
 
+/// An encrypted vote is only valid for specific values of the election public key and crs.
+/// It may be useful to check early if a vote is valid before actually adding it to the tally,
+/// and it is therefore important to verify that it is later added to an encrypted tally that
+/// is consistent with the election public key and crs it was verified against.
```

Comment:
As long as we do this with the right context of the EncryptedTally, it should be sufficient.

Reply:
Let's imagine we have a vote for voteplan A and one for voteplan B. What could happen is that I verify those two perfectly valid votes, but then for some reason they are added to the same `EncryptedTally`.  This is relevant because `Ballot` is used for early verification so those two checks happen at different time and at a different stage in the vote flow, and we need to validate that whatever the program does in the meanwhile does not interfere with the vote validity.

The fingerprint is there to avoid this (again, using a fingerprint instead of full crs and election pk is just for reducing the memory footprint for each vote, not really a concern for an encrypted tally)

Reply:
ah, I see. This makes sense 👍